### DISCUSSION TITLE
Development: Data Source Configuration

### DISCUSSION THREAD

**Discussion by Ka0o0** [renovatebot/renovate#13378] [Request Help] [answered]
Hi,

I'm currently working on implementing a Debian package data source. In Debian, packages for different architectures might differ. To support this, I would need the information of the target platform. Currently, I'm having it as a static default: https://github.com/Ka0o0/renovate/blob/feat/deb-datasource/lib/datasource/deb/index.ts#L38

I thought about making it globally configurable, like this:

```json
{
  "deb": {
    "binaryArch": "amd64"
  }
}
```

What's the best way of achieving this?

Thanks

**Comment by Ka0o0** [comment, marked as answer]
Okay, I've figured out that the datasource's `getReleases` method receives the final version of the configuration which is merged from multiple sources, like the repository root, etc. Despite the data type of the first parameter which is `GetReleasesConfig` the actual object contains all configuration properties.

This means, that I was able to use the global configuration by simply casting this object to an appropriate interface.

**Reply by rarkins** [reply]
Changes to the `getReleases()` options should be done only after great consideration. We should discuss in the apt issue first about the requirements and best way to achieve it
